zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/datastore/IDataConn.scala

```diff
@@ -24,34 +25,37 @@ class AsterixConn(url: String, wSClient: WSClient)(implicit ec: ExecutionContext
 
   override def defaultQueryResponse: JsValue = defaultEmptyResponse
 
-  def postQuery(aql: String): Future[JsValue] = {
-    postWithCheckingStatus(aql, (ws: WSResponse) => ws.json, (ws: WSResponse) => defaultQueryResponse)
+  def postQuery(query: String): Future[JsValue] = {
+    postWithCheckingStatus(query, (ws: WSResponse) => {
+      ws.json.asInstanceOf[JsObject].value("results")
+    }, (ws: WSResponse) => defaultQueryResponse)
   }
 
-  def postControl(aql: String): Future[Boolean] = {
-    postWithCheckingStatus(aql, (ws: WSResponse) => true, (ws: WSResponse) => false)
+  def postControl(query: String): Future[Boolean] = {
+    postWithCheckingStatus(query, (ws: WSResponse) => true, (ws: WSResponse) => false)
   }
 
-  protected def postWithCheckingStatus[T](aql: String, succeedHandler: WSResponse => T, failureHandler: WSResponse => T): Future[T] = {
-    post(aql).map { wsResponse =>
-      if (wsResponse.status == 200) {
+  protected def postWithCheckingStatus[T](query: String, succeedHandler: WSResponse => T, failureHandler: WSResponse => T): Future[T] = {
+    post(query).map { wsResponse =>
+      if (wsResponse.json.asInstanceOf[JsObject].value.get("status") == Some(JsString("success"))) {
         succeedHandler(wsResponse)
-      } else {
-        log.error("AQL failed:" + Json.prettyPrint(wsResponse.json))
+      }
+      else {
+        log.error("Query failed:" + Json.prettyPrint(wsResponse.json))
         failureHandler(wsResponse)
       }
     }
   }
 
-  def post(aql: String): Future[WSResponse] = {
-    log.debug("AQL:" + aql)
-    val f = wSClient.url(url).withRequestTimeout(Duration.Inf).post(aql)
-    f.onFailure(wsFailureHandler(aql))
+  def post(query: String): Future[WSResponse] = {
+    val f = wSClient.url(url).withRequestTimeout(Duration.Inf).post(Map("statement" -> Seq(query), "mode" -> Seq("synchronous"), "include-results" -> Seq("true")))
```

Comment:
could wrap up the `Map("statement" -> Seq(query), "mode" -> Seq("synchronous"), "include-results" -> Seq("true")` to a separate function? 

We may extend that function to extend more *db related settings.